src/plugins/discourse/weights.js

```diff
@@ -1,9 +1,75 @@
 // @flow
 
 import {type NodeWeight} from "../../core/weights";
+import * as C from "../../util/combo";
+import * as NullUtil from "../../util/null";
 import {DEFAULT_TRUST_LEVEL_TO_WEIGHT} from "./createGraph";
 import {type User} from "./fetch";
 
+export type SerializedConfig = {|
+  +defaultWeight?: NodeWeight,
+  +weights?: {
+    [string]: NodeWeight,
+  },
+|};
+
+export type Config = {|
+  +defaultWeight: NodeWeight,
+  +weights: {
+    [string]: NodeWeight,
+  },
+|};
+
+export type CategoryConfig = Config;
```

Comment:
The `categoryConfigParser` ensures that the parsed keys represent
numbers, but that’s not documented on `CategoryConfig` at all. Is that
meant to be a type-level invariant (it’s not, currently), a property of
the parser, or something else?

Reply:
@wchargin It's just meant to be a property of the parser. I'd like it to be a type-level invariant but I ran into a conflict with the `dict` parser where all keys must ultimately be strings, which makes sense.

I realize these types completely lack documentation right now. I was going to go back and document the types after I got everything working end-to-end, but I appreciate your pre-review nonetheless.

Reply:
If we use maps for the configs per my above suggestion, I believe this stops being a problem. The parser can correctly parse the dict keys as numbers where appropriate.